assets/shared/helpers/player.js

```diff
@@ -0,0 +1,557 @@
+		 * @param {number}           seconds The video time in seconds to set.
+		 *
+		 * @return {Promise} A promise that resolves if the video was set to a current time successfully.
+		 */
+		setCurrentTime: ( player, seconds ) => {
+			try {
+				player.currentTime = seconds;
+				return Promise.resolve();
+			} catch ( e ) {
+				return Promise.reject( e );
+			}
+		},
+
+		/**
+		 * Play the video.
+		 *
+		 * @param {HTMLVideoElement} player The player element.
+		 *
+		 * @return {Promise} The native promise from the video play function.
+		 */
+		play: ( player ) => player.play(),
+
+		/**
+		 * Pause the video.
+		 *
+		 * @param {HTMLVideoElement} player The player element.
+		 *
+		 * @return {Promise} A promise that resolves if the video was paused successfully.
+		 */
+		pause: ( player ) => {
+			try {
+				player.pause();
+
+				if ( player.paused ) {
+					return Promise.resolve();
+				}
+
+				return Promise.reject( new Error( "Video didn't pause" ) );
+			} catch ( e ) {
+				return Promise.reject( e );
+			}
+		},
+
+		/**
+		 * Add an event listener to the player.
+		 *
+		 * @param {HTMLVideoElement} player    The player element.
+		 * @param {string}           eventName Event name (currently only `timeupdate` is supported)
+		 * @param {Function}         callback  Listener callback.
+		 *
+		 * @return {Function} The function to unsubscribe the event.
+		 */
+		on: ( player, eventName, callback ) => {
+			const transformedCallback = ( event ) => {
+				callback( event.target.currentTime );
+			};
+
+			player.addEventListener( eventName, transformedCallback );
+
+			return () => {
+				player.removeEventListener( eventName, transformedCallback );
+			};
+		},
+	},
+	[ VIDEOPRESS_TYPE ]: {
+		/**
+		 * The embed pattern to check if it's the respective type.
+		 */
+		embedPattern: /(videopress|video\.wordpress)\.com\/.+/i,
+
+		/**
+		 * Initialize the player.
+		 *
+		 * @param {HTMLIFrameElement} element The player element.
+		 *
+		 * @return {Promise<HTMLIFrameElement>} The video player through a promise.
+		 */
+		initializePlayer: ( element ) =>
+			new Promise( ( resolve ) => {
+				// eslint-disable-next-line @wordpress/no-global-event-listener -- Not in a React context.
+				window.addEventListener( 'message', ( event ) => {
+					if (
+						event.source !== element.contentWindow ||
+						event.data.event !== 'videopress_durationchange' ||
+						! event.data.durationMs
+					) {
+						return;
+					}
+
+					// Set the duration to a dataset in order to have it available for later.
+					element.dataset.duration =
+						parseInt( event.data.durationMs, 10 ) / 1000;
+
+					resolve( element );
+				} );
+			} ),
+
+		/**
+		 * Get the video duration.
+		 *
+		 * @param {HTMLIFrameElement} player The player element.
+		 *
+		 * @return {Promise<number>} The duration of the video in seconds through a promise.
+		 */
+		getDuration: ( player ) => {
+			const duration = player.dataset.duration;
```

Comment:
we could just use 

```suggestion
			const { duration } = player.dataset;
```

right?